python/ray/data/tests/test_operators.py

```diff
@@ -1549,6 +1549,31 @@ def test_per_block_limit_fn(blocks_data, per_block_limit, expected_output):
     assert result_data == expected_output
 
 
+def test_apply_transform_dag_consistency():
+    from ray.data._internal.logical.interfaces.operator import Operator
+
+    a = Operator("A", [])
+    b = Operator("B", [])
+    c = Operator("C", [a, b])
+
+    def transform_b(op: Operator) -> Operator:
+        """Transform only operator B, leaving A unchanged."""
+        if op.name == "B":
+            return Operator("Transformed B", op.input_dependencies)
+        return op
+
+    c_transformed = c._apply_transform(transform_b)
+
+    # This should create the DAG:
+    #               A --|
+    #                   |--> C
+    #   B transformed --|
```

Comment:
Whitspace looks a bit weird?
```suggestion

    # This should create the DAG:
    #               A --|
    #                   |--> C
    #   B transformed --|
    c_transformed = c._apply_transform(transform_b)

```